File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution 
{
    public List<Integer> findDisappearedNumbers(int[] nums) 
    {
        List<Integer> list = new ArrayList<>();
        cyclic(nums);
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i] != i+1)
                list.add(i+1);
        }
        return list;
        
    }
    public void cyclic(int[] nums)
    {
        int i=0;
        while(i<nums.length)
        {
            int correct = nums[i]-1;
            if(nums[i] != nums[correct])
                swap(nums, i, correct);
            else
                i++;
        }
    }
    public void swap(int[] nums,int first,int second)
    {
        int temp = nums[first];
        nums[first] = nums[second];
        nums[second] = temp;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        flag = [0] * len(nums)
        ans=[]
        for i in nums:
            flag[i-1] = 1
        for i in range(0,len(nums)):
            if flag[i] == 0:
                ans.append(i+1)
        return ans